qutip/tests/test_wigner.py

```diff
@@ -593,6 +595,104 @@ def test_wigner_clenshaw_sp_iter_dm():
+    pytest.param(7, id="spin-seven")
+])
+@pytest.mark.parametrize("pure", [
+    pytest.param(True, id="pure"),
+    pytest.param(False, id="mixed")
+])
+def test_spin_q_function(spin, pure):
+    d = int(2 * spin + 1)
+    rho = rand_dm(d, pure=pure)
+
+    # Points at which to evaluate the spin Q function
+    theta = np.linspace(0, np.pi, 32, endpoint=True)
+    phi = np.linspace(-np.pi, np.pi, 64, endpoint=True)
+    Q, _, _ = qutip.spin_q_function(rho, theta, phi)
+
+    for k, (phi_prime, theta_prime) in enumerate(itertools.product(phi, theta)):
+        state = qutip.spin_coherent(spin, theta_prime, phi_prime)
+        direct_Q = (2 * spin + 1) / (4 * np.pi) * (state.dag() * rho * state).norm()
+        assert_almost_equal(Q.flat[k], direct_Q, decimal=9)
+
+@pytest.mark.parametrize(['spin'], [
+    pytest.param(1/2, id="spin-one-half"),
+    pytest.param(3, id="spin-three"),
+    pytest.param(13/2, id="spin-thirteen-half"),
+    pytest.param(7, id="spin-seven")
+])
+@pytest.mark.parametrize("pure", [
+    pytest.param(True, id="pure"),
+    pytest.param(False, id="mixed")
+])
+def test_spin_q_function_normalized(spin, pure):
+    d = int(2 * spin + 1)
+    rho = rand_dm(d, pure=pure)
+
+    # Points at which to evaluate the spin Q function
+    theta = np.linspace(0, np.pi, 512, endpoint=True)
+    phi = np.linspace(-np.pi, np.pi, 1024, endpoint=True)
+    Q, THETA, _ = qutip.spin_q_function(rho, theta, phi)
+
+    norm = np.trapz(np.trapz(Q * np.sin(THETA), theta), phi)
+    assert_almost_equal(norm, 1, decimal=5)
+
+
+@pytest.mark.parametrize(["spin"], [
+    pytest.param(1/2, id="spin-one-half"),
+    pytest.param(3, id="spin-three"),
+    pytest.param(13/2, id="spin-thirteen-half"),
+    pytest.param(7, id="spin-seven")
+])
+@pytest.mark.parametrize("pure", [
+    pytest.param(True, id="pure"),
+    pytest.param(False, id="mixed")
+])
+def test_spin_wigner_normalized_and_real(spin, pure):
+    d = int(2 * spin + 1)
+    rho = rand_dm(d, pure=pure)
+
+    # Points at which to evaluate the spin Wigner function
+    theta = np.linspace(0, np.pi, 256, endpoint=True)
+    phi = np.linspace(-np.pi, np.pi, 512, endpoint=True)
+    W, THETA, PHI = qutip.spin_wigner(rho, theta, phi)
+
+    assert_allclose(W.imag, 0, atol=1e-9)
+
+    norm = np.trapz(np.trapz(W * np.sin(THETA), theta), phi)
+    assert_almost_equal(norm, 1, decimal=4)
+
+@pytest.mark.parametrize(['spin'], [
+    pytest.param(1/2, id="spin-one-half"),
+    pytest.param(3, id="spin-three"),
+    pytest.param(13/2, id="spin-thirteen-half"),
+    pytest.param(7, id="spin-seven")
+])
+@pytest.mark.parametrize("pure", [
+    pytest.param(True, id="pure"),
+    pytest.param(False, id="mixed")
+])
+def test_spin_wigner_overlap(spin, pure, n=10):
+    d = int(2*spin + 1)
+    rho = rand_dm(d, pure=pure)
+
+    # Points at which to evaluate the spin Wigner function
+    theta = np.linspace(0, np.pi, 256, endpoint=True)
+    phi = np.linspace(-np.pi, np.pi, 512, endpoint=True)
```

Comment:
```suggestion
    theta = np.linspace(0, np.pi, 128, endpoint=True)
    phi = np.linspace(-np.pi, np.pi, 256, endpoint=True)
```
This was faster for me, but still passed the accuracy threshold during repeated test runs.

Reply:
Okay to leave these as is for now.